mpxe/sims/mcp23008.py

```diff
@@ -0,0 +1,32 @@
+from mpxe.protogen import stores_pb2
+from mpxe.protogen import mcp23008_pb2
+
+import time
+
+from mpxe.sims import sim
+
+MCP23008_KEY = (stores_pb2.MxStoreType.MCP23008, 0)
+NUM_MCP_PINS = 8
+
+class Mcp23008(sim.Sim):
+    
+    def handle_update(self, pm, proj):
+        stores = proj.stores
+        if MCP23008_KEY in stores:
+            mcp         = stores[MCP23008_KEY]
+            self.states  = [bool(mcp.state[i]) for i in range(len(mcp.state)) if i < NUM_MCP_PINS]
+
+    # Update the store with a new pin state
+    def update_pin_state(self, proj, pin, state):
```

Comment:
This isn't used in the smoketest, but I figured it will be needed if we ever to simulate any inputs from an mcp23008